Library/Homebrew/cask/cask.rb

```diff
@@ -166,13 +166,11 @@ def outdated_versions(greedy: false, greedy_latest: false, greedy_auto_updates:
         version
       end
 
-      if greedy || greedy_latest || (greedy_auto_updates && auto_updates)
-        if latest_version.latest?
-          return versions if outdated_download_sha?
+      if latest_version.latest?
+        return versions if (greedy || greedy_latest) && outdated_download_sha?
 
-          return []
-        end
-      elsif auto_updates
+        return []
+      elsif auto_updates && !(greedy || greedy_auto_updates)
         return []
```

Comment:
```suggestion
      elsif latest_version.latest? || 
            (auto_updates && !(greedy || greedy_auto_updates))
        return []
```

Reply:
```ruby
!(greedy || greedy_autoupdates) == !greedy && !greedy_autoupdates
```
The right-hand side is easier to parse, IMHO.